.github/workflows/main.yml

```diff
@@ -40,6 +40,10 @@ jobs:
         uses: actions/checkout@v3
         with:
           submodules: recursive
+      - name: Build Veracruz-Linux
+        id: linux-build
+        run: |
+           make -C /work/veracruz/workspaces linux
```

Comment:
Would it make sense to add `PROFILE=release`?

Reply:
Hmm wouldn't that significantly slow down the build?

Reply:
I've just tried the `make linux` and `make linux-tests` with and without `PROFILE=release` (on both commands) on my machine:
without: 4:21 + 8:27
with: 7:55 + 2:17
Is that plausible, or was something else interfering, I wonder?
If it is roughly representative, then `PROFILE=release` is worth having even without the `vod-full-deployment`.
Perhaps I'll set it to run a few more times while I'm having lunch ...

Reply:
Top tip: Don't try to use `time` in a `/bin/sh` script!

Reply:
Ooh interesting! Can you reproduce that experiment a few times?
Maybe we could also limit the build to the artifacts we need: `make -C /work/veracruz/workspaces linux` builds the Rust examples and data generators, which we don't need

Reply:
We decided to stick with a debug build for the CI because the GitHub machine doesn't have lots of cores for parallel building, but since I have the data from my machine I might as well post it here. I've ordered the times in each line so the third one is the median of the five.
```
debug build real: 3m24.437s, 3m25.781s, 3m26.476s, 3m28.335s, 3m29.153s
debug build user: 26m0.750s, 26m10.264s, 26m18.469s, 26m3.376s, 26m6.292s
debug build sys: 2m24.606s, 2m24.846s, 2m24.939s, 2m25.214s, 2m27.750s

debug tests real: 6m13.795s, 6m20.578s, 6m22.794s, 6m24.346s, 6m25.020s
debug tests user: 4m21.314s, 4m21.385s, 4m21.557s, 4m22.661s, 4m24.407s
debug tests sys: 0m26.566s, 0m26.674s, 0m26.951s, 0m27.113s, 0m27.270s

release build real: 7m46.685s, 7m46.804s, 7m48.685s, 7m49.407s, 7m49.688s
release build user: 37m19.445s, 37m3.408s, 37m7.533s, 37m8.536s, 37m9.781s
release build sys: 2m1.333s, 2m1.609s, 2m1.728s, 2m2.033s, 2m3.264s

release tests real: 2m12.240s, 2m12.978s, 2m8.444s, 2m8.673s, 2m8.675s
release tests user: 3m36.855s, 3m36.911s, 3m37.511s, 3m38.965s, 3m39.464s
release tests sys: 0m9.687s, 0m9.763s, 0m9.815s, 0m9.870s, 0m9.946s
```

Reply:
Interesting! How come debug tests is slower than release tests?

Reply:
Debug builds are faster because they do less optimizations and no LTO. The resulting binaries are also slower, often much slower.

Reply:
@mathias-arm Never mind, I misread @egrimley-arm 's table